modules/imap/site.js

```diff
@@ -757,29 +769,64 @@ var imap_message_view_finished = function(msg_uid, detail, skip_links) {
     }
     if (detail && !skip_links) {
         Hm_Message_List.prev_next_links(msg_uid);
+        class_name = 'imap_' + detail.server_id + '_' + msg_uid + '_' + detail.folder;
+        if (hm_list_parent() === 'combined_inbox') {
+            Hm_Message_List.prev_next_links('formatted_combined_inbox', class_name);
+        }
+        else if (hm_list_parent() === 'unread') {
+            Hm_Message_List.prev_next_links('formatted_unread_data', class_name);
+        }
+        else if (hm_list_parent() === 'flagged') {
+            Hm_Message_List.prev_next_links('formatted_flagged_data', class_name);
+        }
+        else if (hm_list_parent() === 'advanced_search') {
+            Hm_Message_List.prev_next_links('formatted_advanced_search_data', class_name);
+        }
+        else if (hm_list_parent() === 'search') {
+            Hm_Message_List.prev_next_links('formatted_search_data', class_name);
+        }
+        else if (hm_list_parent() === 'sent') {
+            Hm_Message_List.prev_next_links('formatted_sent_data', class_name);
+        }
+        else if (hm_list_parent() === 'junk') {
+            Hm_Message_List.prev_next_links('formatted_junk_data', class_name);
+        }
+        else if (hm_list_parent() === 'trash') {
+            Hm_Message_List.prev_next_links('formatted_trash_data', class_name);
+        }
+        else if (hm_list_parent() === 'drafts') {
+            Hm_Message_List.prev_next_links('formatted_drafts_data', class_name);
+        }
+        else if (hm_list_parent() === 'tag') {
+            Hm_Message_List.prev_next_links('formatted_tag_data', class_name);
+        }
+        else {
+            var key = 'imap_' + Hm_Utils.get_url_page_number() + '_' + getListPathParam();
+            Hm_Message_List.prev_next_links(key, class_name);
+        }
```

Comment:
This rebase isn't accurate. Master has passed over these changes.

Reply:
@amaninyumu1 I think you should be careful merging the changes from master - there are conflicts because of the upstream changes and these need to be merged correctly (keep the master ones and copy your code to the relevant places).